packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -69,29 +76,39 @@ export class BlobManager {
         });
     }
 
+    private hasBlob(id: string): boolean {
+        return this.blobIds.has(id) || this.pendingBlobIds.has(id) ||
+            this.detachedBlobIds.has(id) || !!this.redirectTable?.has(id);
+    }
+
     public async getBlob(blobId: string): Promise<IFluidHandle<ArrayBufferLike>> {
-        assert(this.blobIds.has(blobId) || this.pendingBlobIds.has(blobId), 0x11f /* "requesting unknown blobs" */);
+        assert(this.hasBlob(blobId), 0x11f /* "requesting unknown blobs" */);
+
+        const storageId = this.redirectTable?.get(blobId) ?? blobId;
         return new BlobHandle(
-            `${BlobManager.basePath}/${blobId}`,
+            `${BlobManager.basePath}/${storageId}`,
             this.routeContext,
-            async () => this.getStorage().readBlob(blobId),
+            async () => this.getStorage().readBlob(storageId),
         );
     }
 
     public async createBlob(blob: ArrayBufferLike): Promise<IFluidHandle<ArrayBufferLike>> {
-        assert(
-            this.runtime.attachState !== AttachState.Attaching,
-            0x1f9 /* "createBlob() while attaching not supported" */,
-        );
-        const response = await this.getStorage().createBlob(blob);
+        if (this.runtime.attachState === AttachState.Attaching) {
+            // blob upload is not supported in "Attaching" state
+            await new Promise<void>((res) => this.runtime.once("attached", res));
```

Comment:
Consider adding telemetry event here such that we know how often it happens and either have a chance to revisit this area or advice customer not to change their design